File: LanPlatform/News/WeatherStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using GabionPlatform.Database;

namespace GabionPlatform.News
{
    public class WeatherStatus : DatabaseObject
    {
        public String Source { get; set; }

        // Current
        public int CurrentTemperature { get; set; }
        public WeatherType CurrentWeatherType { get; set; }
        public int CurrentRainChance { get; set; }
        public long CurrentTime { get; set; }

        // Future
        public int FirstTemperature { get; set; }
        public WeatherType FirstWeatherType { get; set; }
        public int FirstRainChance { get; set; }
        public long FirstTime { get; set; }

        public int SecondTemperature { get; set; }
        public WeatherType SecondWeatherType { get; set; }
        public int SecondRainChance { get; set; }
        public long SecondTime { get; set; }

        public int ThirdTemperature { get; set; }
        public WeatherType ThirdWeatherType { get; set; }
        public int ThirdRainChance { get; set; }
        public long ThirdTime { get; set; }

        // Daily
        public int DailyRainChance { get; set; }
        public int DailyHigh { get; set; }
        public int DailyLow { get; set; }

        public long Sunrise { get; set; }
        public long Sunset { get; set; }

        public WeatherStatus()
        {
            Source = "";

            CurrentTemperature = 0;
            CurrentWeatherType = WeatherType.None;
            CurrentRainChance = 0;
            CurrentTime = 0;

            FirstTemperature = 0;
            FirstWeatherType = WeatherType.None;
            FirstRainChance = 0;
            FirstTime = 0;

            SecondTemperature = 0;
            SecondWeatherType = WeatherType.None;
            SecondRainChance = 0;
            SecondTime = 0;

            ThirdTemperature = 0;
            ThirdWeatherType = WeatherType.None;
            ThirdRainChance = 0;
            ThirdTime = 0;

            DailyRainChance = 0;
            DailyHigh = 0;
            DailyLow = 0;

            Sunrise = 0;
            Sunset = 0;
        }
    }

    public enum WeatherType
    {
        None = 0,
        Sunny,
        Cloudy,
        Rain,
        Lightning,
        Snow
    }
}

File: LanPlatform/platformJS/News/WeatherStatus.js
LPNews.WeatherStatus = function() {
    // Current
    this.CurrentTemperature = 0;
    this.CurrentWeatherType = 0;
    this.CurrentRainChance = 0;
    this.CurrentTime = 0;

    // Future
    this.FirstTemperature = 0;
    this.FirstWeatherType = 0;
    this.FirstRainChance = 0;
    this.FirstTime = 0;

    this.SecondTemperature = 0;
    this.SecondWeatherType = 0;
    this.SecondRainChance = 0;
    this.SecondTime = 0;

    this.ThirdTemperature = 0;
    this.ThirdWeatherType = 0;
    this.ThirdRainChance = 0;
    this.ThirdTime = 0;

    // Daily
    this.DailyRainChance = 0;
    this.DailyHigh = 0;
    this.DailyLow = 0;

    this.Sunrise = 0;
    this.Sunset = 0;
}

LPNews.WeatherStatus.prototype.LoadModel = function(model) {
    // Current
    this.CurrentTemperature = model.CurrentTemperature;
    this.CurrentWeatherType = model.CurrentWeatherType;
    this.CurrentRainChance = model.CurrentRainChance;
    this.CurrentTime = model.CurrentTime;

    // Future
    this.FirstTemperature = model.FirstTemperature;
    this.FirstWeatherType = model.FirstWeatherType;
    this.FirstRainChance = model.FirstRainChance;
    this.FirstTime = model.FirstTime;

    this.SecondTemperature = model.SecondTemperature;
    this.SecondWeatherType = model.SecondWeatherType;
    this.SecondRainChance = model.SecondRainChance;
    this.SecondTime = model.SecondTime;

    this.ThirdTemperature = model.ThirdTemperature;
    this.ThirdWeatherType = model.ThirdWeatherType;
    this.ThirdRainChance = model.ThirdRainChance;
    this.ThirdTime = model.ThirdTime;

    // Daily
    this.DailyRainChance = model.DailyRainChance;
    this.DailyHigh = model.DailyHigh;
    this.DailyLow = model.DailyLow;

    this.Sunrise = model.Sunrise;
    this.Sunset = model.Sunset;

    return;
}

LPNews.WeatherStatus.prototype.GetFirstTime = function() {
    var a = new Date(this.FirstTime * 1000);

    var hour = a.getHours();
    var min = a.getMinutes();
    var suffix = (hour >= 12) ? " PM" : " AM";

    hour = hour % 12;

    if (hour == 0) {
        hour = 12;
    }

    var time = hour + ((min < 10) ? ":0" : ":") + min + suffix;

    return time;
}